crates/platform-tags/src/tags.rs

```diff
@@ -358,8 +358,12 @@ fn compatible_tags(platform: &Platform) -> Result<Vec<String>, PlatformError> {
             let mut platform_tags = vec![format!("linux_{}", arch)];
             if let Some(min_minor) = arch.get_minimum_manylinux_minor() {
                 platform_tags.extend(
-                    (min_minor..=*minor).map(|minor| format!("manylinux_{major}_{minor}_{arch}")),
+                    (min_minor..=*minor)
+                        .map(|minor| format!("manylinux_{major}_{minor}_{arch}"))
+                        .rev(),
                 );
+                // Support legacy manylinux tags with lower priority
+                // <https://peps.python.org/pep-0600/#legacy-manylinux-tags>
                 if (min_minor..=*minor).contains(&12) {
                     platform_tags.push(format!("manylinux2010_{arch}"));
```

Comment:
Doesn't `manylinux2014` need to be greater than `manylinux2010`, and also greater than `manylinux_2_12`? I think these legacy tags need to be interspersed.

Reply:
Eek okay this was more wrong than I thought to start..